contracts/staking/feirari/AutoRewardsDistributorV2.sol

```diff
@@ -0,0 +1,126 @@
+pragma solidity ^0.8.0;
+
+import "./../ITribalChief.sol";
+import "../../refs/CoreRef.sol";
+import "../../external/Unitroller.sol";
+import "../StakingTokenWrapper.sol";
+import "./IRewardsDistributorAdmin.sol";
+
+/// @notice Controller Contract to set tribe per block in Rewards Distributor Admin on Rari
+contract AutoRewardsDistributorV2 is CoreRef {
```

Comment:
What's the difference between this contract and the original ARD?

Reply:
The diff is in the PR description

Reply:
The point is that it can now be deployed without knowing either the exact tribalChief index or the ctoken address.

The ctoken part is particularly important because ctokens can only be deployed directly by the comptroller via the admin (OA) and it includes the block number in the address salt